internal_displacement/interpreter.py

```diff
@@ -110,21 +111,58 @@ def get_absolute_date(relative_date_string, publication_date=None):
     -----------
     relative_date_string        the relative date in an article (e.g. 'Last week'): String
     publication_date            the publication_date of the article: datetime
+    
     Returns:
     --------
     One of: 
         - a datetime that represents the absolute date of the relative date based on 
             the publication_date
         - None, if parse is not successful
-        TODO: Q: In this case, should we return the publication_date instead?
     """
 
     cal = parsedatetime.Calendar()
     parsed_result = cal.nlp(relative_date_string, publication_date)
     if parsed_result is not None:
-        # Check if parse is successful
-        return parsed_result[0][0]
+        # Parse is successful
+        parsed_absolute_date = parsed_result[0][0]
+
+        # Assumption: input date string is in the past
+        # If parsed date is in the future (relative to publication_date), 
+        #   we roll it back to the past
+        
+        if publication_date and parsed_absolute_date > publication_date:
+            # parsedatetime returns a date in the future
+            # likely because year isn't specified or date_string is relative
+            
+            # Check a specific date is included
+            # TODO: Smarter way or regex to check if relative_date_string 
+            #       contains a month name?
+            months = ['jan', 'feb', 'mar', 'apr', 'may', 'jun', 
```

Comment:
Is there a better way to test if a ```date string``` specifies a particular date (e.g. '16th December' vs 'a week ago')? I don't really want to use another library just for this test...

If we are going to stick with this, should we move this somewhere else? (in case we need to handle non-english articles)